lightning/src/routing/router.rs

```diff
@@ -150,39 +158,161 @@ impl cmp::PartialOrd for RouteGraphNode {
 struct DummyDirectionalChannelInfo {
 	cltv_expiry_delta: u32,
 	htlc_minimum_msat: u64,
+	htlc_maximum_msat: Option<u64>,
 	fees: RoutingFees,
 }
 
+/// It's useful to keep track of the hops associated with the fees required to use them,
+/// so that we can choose cheaper paths (as per Dijkstra's algorithm).
+/// Fee values should be updated only in the context of the whole path, see update_value_and_recompute_fees.
+/// These fee values are useful to choose hops as we traverse the graph "payee-to-payer".
+#[derive(Clone)]
+struct PathBuildingHop {
+	/// Hop-specific details unrelated to the path during the routing phase,
+	/// but rather relevant to the LN graph.
+	route_hop: RouteHop,
+	/// Minimal fees required to route to the source node of the current hop via any of its inbound channels.
+	src_lowest_inbound_fees: RoutingFees,
+	/// Fees of the channel used in this hop.
+	channel_fees: RoutingFees,
```

Comment:
How come `base_msat` doesn't matter here? Both base and proportional come from directional info, which comes from gossip, so they are equal?

Reply:
I agree `base_msat` matters, I just wanted to underscore that `RoutingFees` isn't a _quantity_ but rather a _ratio_. It would be a quantity if `base_msat` was the only factor but it's not.